devops/providers/azure-devops/templates/infrastructure/azure-pipeline-build-stage.yml

```diff
@@ -6,6 +6,7 @@ jobs:
 
 - job: TemplateChangeDetection
   displayName: Determine CI Targets to Run
+  pool: $(AGENT_POOL)
```

Comment:
Adding this, will it cause us to only use custom build agents? Can we still use hosted agents? 

We don't want every customer/template to use custom build agents only, correct?

Reply:
Follow on -- have we tested with hosted and non hosted agents? I recall there being some nuances in how the `deployment` step is configured with regard to build agents

Reply:
Great points. I’ll find out. I didn’t try the hosted agent, after this change.

Reply:
@iphilpot if `AGENT_POOL` is set to `Hosted Ubuntu 1604` then Cobalt will use hosted agents, as before. 

@nmiodice @iphilpot I've tested the hosted agents here (an AzDO Pipeline build):
https://dev.azure.com/sthende/sthende-291/_build/results?buildId=52&view=results